exchanges/gateio/gateio.go

```diff
@@ -421,169 +933,2821 @@ func (g *Gateio) SendAuthenticatedHTTPRequest(ctx context.Context, ep exchange.U
+		return errInvalidOrEmptySubaccount
+	}
+	arg.Direction = strings.ToLower(arg.Direction)
+	if arg.Direction != "to" && arg.Direction != "from" {
+		return errInvalidTransferDirection
+	}
+	if arg.Amount <= 0 {
+		return errInvalidAmount
+	}
+	if arg.SubAccountType != "" && arg.SubAccountType != asset.Spot.String() && arg.SubAccountType != asset.Futures.String() && arg.SubAccountType != asset.CrossMargin.String() {
+		return fmt.Errorf("%v; only %v,%v, and %v are allowed", errInvalidAssetType, asset.Spot, asset.Futures, asset.CrossMargin)
+	}
+	return g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, walletSubAccountTransfer, nil, &arg, nil)
+}
+
+// GetSubAccountTransferHistory retrieve transfer records between main and sub accounts.
+// retrieve transfer records between main and sub accounts. Record time range cannot exceed 30 days
+// Note: only records after 2020-04-10 can be retrieved
+func (g *Gateio) GetSubAccountTransferHistory(ctx context.Context, subAccountUserID string, from, to time.Time, offset, limit uint64) ([]SubAccountTransferResponse, error) {
+	params := url.Values{}
+	if subAccountUserID != "" {
+		params.Set("sub_uid", subAccountUserID)
+	}
+	startingTime, err := time.Parse("2006-Jan-02", "2020-Apr-10")
+	if err != nil {
+		return nil, err
+	}
+	if err := common.StartEndTimeCheck(startingTime, from); err == nil {
+		params.Set("from", strconv.FormatInt(from.Unix(), 10))
+	}
+	if err := common.StartEndTimeCheck(from, to); err == nil {
+		params.Set("to", strconv.FormatInt(to.Unix(), 10))
+	}
+	if offset > 0 {
+		params.Set("offset", strconv.FormatUint(offset, 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	var response []SubAccountTransferResponse
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot,
+		http.MethodGet, walletSubAccountTransfer, params, nil, &response)
+}
+
+// GetWithdrawalStatus retrieves withdrawal status
+func (g *Gateio) GetWithdrawalStatus(ctx context.Context, ccy currency.Code) ([]WithdrawalStatus, error) {
+	params := url.Values{}
+	if !ccy.IsEmpty() {
+		params.Set("currency", ccy.String())
+	}
+	var response []WithdrawalStatus
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, walletWithdrawStatus, params, nil, &response)
+}
+
+// GetSubAccountBalances retrieve sub account balances
+func (g *Gateio) GetSubAccountBalances(ctx context.Context, subAccountUserID string) ([]SubAccountBalance, error) {
+	params := url.Values{}
+	if subAccountUserID != "" {
+		params.Set("sub_uid", subAccountUserID)
+	}
+	var response []SubAccountBalance
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, walletSubAccountBalance, params, nil, &response)
+}
+
+// GetSubAccountMarginBalances query sub accounts' margin balances
+func (g *Gateio) GetSubAccountMarginBalances(ctx context.Context, subAccountUserID string) ([]SubAccountMarginBalance, error) {
```

Comment:
Locked is a string, not a float64
https://www.gate.io/docs/developers/apiv4/en/#query-sub-accounts-margin-balances

Reply:
It represents a **locked amount** a numerical value in a string as of borrowed, available, and interest. That's why I declared it as a floating number.